websites/W/WebProfiles/presence.ts

```diff
@@ -4,25 +4,39 @@ const presence = new Presence({
 	browsingTimestamp = Math.floor(Date.now() / 1000);
 
 presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "webprofiles_logo",
+	};
+
 	const showTimestamp = await presence.getSetting<boolean>("timestamp"),
 		showButtons = await presence.getSetting<boolean>("buttons"),
-		presenceData: PresenceData = {
-			largeImageKey: "webprofiles_logo",
+		pages: Record<string, PresenceData> = {
+			"/about/team": { details: "Viewing page:", state: "Team" },
+			"/about/partners": { details: "Viewing page:", state: "Partners" },
+			"/discover": { details: "Viewing page:", state: "Discover" },
+			"/@me/settings": { details: "Editing my profile" },
+			"/login": { details: "Logging in..." },
+			register: { details: "Registering..." },
 		};
 
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.pathname.includes(path))
```

Comment:
Properties on `document.location` are used often. For consistency/simplicity, it may be a good idea to destructure them.